src/core/include/openvino/op/unique.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/op.hpp"
+
+namespace ov {
+namespace op {
+namespace v10 {
+
+/// \brief Operator which selects and returns unique elements or unique slices of the input tensor
+/// \ingroup ov_ops_cpp_api
+class OPENVINO_API Unique : public Op {
```

Comment:
Please add `Unique` op to `OpImplCheckTest`
https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/single_op_graph.cpp